shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/Apple.kt

```diff
@@ -63,19 +64,35 @@ class AppleConfigurablesImpl(
         XcodePartsProvider.InternalServer
     } else {
         val xcode = Xcode.current
-        properties.getProperty("useFixedXcodeVersion")?.let { requiredXcodeVersion ->
-            val currentXcodeVersion = xcode.version
 
-            if (properties.getProperty("ignoreXcodeVersionCheck") != "true" &&
-                    currentXcodeVersion != requiredXcodeVersion) {
-                error("expected Xcode version $requiredXcodeVersion, got $currentXcodeVersion, consider updating " +
-                        "Xcode or use \"ignoreXcodeVersionCheck\" variable in konan.properties")
+        if (properties.getProperty("ignoreXcodeVersionCheck") != "true") {
+            properties.getProperty("minimalXcodeVersion")?.let { minimalXcodeVersion ->
+                val currentXcodeVersion = xcode.version
+                checkXcodeVersion(minimalXcodeVersion, currentXcodeVersion)
             }
         }
 
         XcodePartsProvider.Local(xcode)
     }
 
+    private fun checkXcodeVersion(minimalVersion: String, currentVersion: String) {
+        val minimalVersionParts = minimalVersion.split('.').map { it.toInt() }
```

Comment:
Cannot Xcode version contain non-number parts?

Reply:
I've checked a beta version and it reports `11.2`. Just in case made the check more stable.